docs/UserGuide.md

```diff
@@ -131,22 +131,24 @@ Examples:
 
 ### Locating contacts by name : `find`
 
-Finds contacts whose names contain any of the given keywords.
+Finds contacts whose details matches all given fields.
 
-Format: `find KEYWORD [MORE_KEYWORDS]`
+Format: `find [n/NAME] [th/TELEGRAM_HANDLE] [e/EMAIL] [a/ADDRESS] [r/ROLE]…​ [nn/NICKNAME]`
 
-* The search is case-insensitive. e.g `hans` will match `Hans`
-* The order of the keywords does not matter. e.g. `Hans Bo` will match `Bo Hans`
-* Only the name is searched.
-* Substrings of the name will also be matched e.g. Query `Han` will match `Hans`
-* Contacts matching at least one keyword will be returned (i.e. `OR` search).
-  e.g. `Hans Bo` will return `Hans Gruber`, `Bo Yang`
+* The role field(s), if any, have to take a valid role value (Refer to **Tip** section in [adding a contact](#adding-a-contact-add))
+* The search for all other fields is case-insensitive. e.g. query `hans` will match `Hans`
+* The order of the words does not matter. e.g. query `Hans Bo` will match `Bo Hans`
+* Queries that are substrings of details will also be matched e.g. query `Han` will match `Hans`
+  * Each word is treated as a separate substring e.g. query `Hans Ansbo` will match `Hansbo`
```

Comment:
Each keyword?
Let's standardise if the terms in the find command are called keywords or word.

`find n/Alex Sunny Tan r/ President r/ Vice President`
Hold on I am getting a bit confused on the criteria allowed for finding, let me submit the review first then comment later after I internalise the user guide.